src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -0,0 +1,435 @@
+_CONFIG_FOR_DOC = "ResNetConfig"
+_FEAT_EXTRACTOR_FOR_DOC = "ConvNextFeatureExtractor"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = ""
+_EXPECTED_OUTPUT_SHAPE = [1, 2048, 7, 7]
+
+# Image classification docstring
+_IMAGE_CLASS_CHECKPOINT = ""
+_IMAGE_CLASS_EXPECTED_OUTPUT = "'tabby, tabby cat'"
+
+RESNET_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Francesco/resnet50",
+    # See all resnet models at https://huggingface.co/models?filter=resnet
+]
+
+
+class ResNetConvLayer(nn.Sequential):
+    def __init__(
+        self, in_channels: int, out_channels: int, kernel_size: int = 3, stride: int = 1, activation: str = "relu"
+    ):
+        super().__init__()
+        self.conv = nn.Conv2d(
+            in_channels, out_channels, kernel_size=kernel_size, stride=stride, padding=kernel_size // 2, bias=False
+        )
+        self.bn = nn.BatchNorm2d(out_channels)
+        self.act = ACT2FN[activation] if activation is not None else nn.Identity()
+
+
+class ResNetEmbeddings(nn.Sequential):
+    """
+    ResNet Embedddings (stem) composed of a single aggressive convolution.
+    """
+
+    def __init__(self, num_channels: int, out_channels: int, activation: str = "relu"):
+        super().__init__()
+        self.embedder = ResNetConvLayer(num_channels, out_channels, kernel_size=7, stride=2, activation=activation)
+        self.pool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
+
+
+class ResNetShortCut(nn.Sequential):
+    """
+    ResNet shortcut, used to project the residual features to the correct size. If needed, it is also used to
+    downsample the input using `stride=2`.
+    """
+
+    def __init__(self, in_channels: int, out_channels: int, stride: int = 2):
+        super().__init__()
+        self.conv = nn.Conv2d(in_channels, out_channels, kernel_size=1, stride=stride, bias=False)
+        self.bn = nn.BatchNorm2d(out_channels)
+
+
+class ResNetBasicLayer(nn.Module):
+    """
+    A classic ResNet's residual layer composed by a two `3x3` convolutions.
+
+    Args:
+        in_channels (`int`):
+            The number of input channels.
+        out_channels (`int`):
+            The number of outputs channels.
+        stride (`int`, *optional*, defaults to 1):
+            The stride used in the first convolution.
+        activation (`int`, *optional*, defaults to `"relu"`):
+            The activation used by the layer.
+    """
+
+    def __init__(self, in_channels: int, out_channels: int, stride: int = 1, activation: str = "relu"):
+        super().__init__()
+        should_apply_shortcut = in_channels != out_channels or stride != 1
+        self.shortcut = (
+            ResNetShortCut(in_channels, out_channels, stride=stride) if should_apply_shortcut else nn.Identity()
+        )
+        self.layer = nn.Sequential(
+            ResNetConvLayer(in_channels, out_channels, stride=stride),
+            ResNetConvLayer(out_channels, out_channels, activation=None),
+        )
+        self.activation = ACT2FN[activation]
```

Comment:
I see you have a great docstring here, which describes the expected types for each argument.

Additionnally, you have type annotations for each argument. 

Imo, this is duplication, and harms readability. I would go for one of both, but not both (I'll prefer the docstring).

Similarly, I would update the 3 classes above: ResNetConvLayer, ResNetEmbeddings and ResNetShortCut to also adopt either a docstring or a type annotation.

Reply:
well, types are for developers meaning that you can see it when hovering the function/class in your IDE. Let me know your thoughts

Reply:
I'd advocate to either use docstrings or types, not both as is the case here.

Reply:
docstring removed